synapse/handlers/room.py

```diff
@@ -1195,17 +1195,10 @@ async def filter_evts(events: List[EventBase]) -> List[EventBase]:
 
         # Fetch the aggregations.
         aggregations = await self.store.get_bundled_aggregations(
-            [event], user.to_string()
-        )
-        aggregations.update(
-            await self.store.get_bundled_aggregations(
-                results["events_before"], user.to_string()
-            )
-        )
-        aggregations.update(
-            await self.store.get_bundled_aggregations(
-                results["events_after"], user.to_string()
-            )
+            itertools.chain(
+                results["events_before"], (event,), results["events_after"]
+            ),
+            user.to_string(),
```

Comment:
It avoids creating a couple dictionaries to then `update()` but is mostly clean-up.

Once the changes for  #11825 land than this *is* a performance gain since fewer queries will occur.

Reply:
(It wasn't really related to the open PRs for #11825, so I put it in here since I was touching this code anyway, sorry about that!)

Reply:
Thanks, that makes sense. And not to worry, I think it makes sense here.

Reply:
Today's review is sponsored by the phrase "makes sense".